community/salt/APKBUILD

```diff
@@ -1,67 +1,166 @@
-pkgrel=0
-pkgdesc="A remote execution and communication system built on zeromq"
+pkgver=2016.3.3
+pkgrel=1
+pkgdesc="A parallel remote execution system"
 url="https://github.com/saltstack/salt"
 arch="noarch"
 license="ASL 2.0"
-depends="py-tornado py2-markupsafe py2-futures" 
+depends="py-tornado py2-yaml py-jinja2 py2-markupsafe py2-msgpack py2-crypto py2-zmq
+    py2-six py2-requests procps"
 depends_dev=""
-makedepends="$depends_dev python2-dev py-zmq
-	py2-crypto py2-psutil py-jinja2 py-yaml py-msgpack"
+makedepends="$depends_dev python2-dev py-tornado py2-yaml py-jinja2 
+    py2-markupsafe py2-msgpack py2-crypto py2-zmq py2-requests py2-futures"
 install=""
-subpackages="$pkgname-doc"
-source="https://github.com/saltstack/salt/releases/download/v$pkgver/$pkgname-$pkgver.tar.gz
-		salt-master.initd
-		salt-master.confd
-		salt-minion.initd
-		salt-minion.confd
-		salt-syndic.initd
-		salt-syndic.confd"
+subpackages="$pkgname-doc $pkgname-master $pkgname-minion $pkgname-syndic
+    $pkgname-api $pkgname-cloud $pkgname-ssh"
+source="$pkgname-$pkgver.tar.gz::https://codeload.github.com/saltstack/$pkgname/tar.gz/v$pkgver
+        salt-api.confd
+        salt-api.initd
+        salt-master.confd
+        salt-master.initd
+        salt-minion.confd
+        salt-minion.initd
+        salt-syndic.confd
+        salt-syndic.initd
+        0001-alpine-support.patch"
 
 builddir="$srcdir"/$pkgname-$pkgver
+prepare() {
+    local i
+    cd "$builddir"
+    for i in $source; do
+        case $i in
+        *.patch) msg $i; patch -p1 -i "$srcdir"/$i || return 1;;
+        esac
+    done
+}
 
 build() {
-	cd "$builddir"
-	python2 setup.py build || return 1
+    cd "$builddir"
+    python2 setup.py build || return 1
 }
 
 package() {
-	cd "$builddir"
-	python2 setup.py install --root=$pkgdir/ --optimize=1 || return 1
-	mkdir -p ${pkgdir}/etc/rc.d/
-	cp ${srcdir}/salt-master.initd ${pkgdir}/etc/rc.d/salt-master
-	cp ${srcdir}/salt-minion.initd ${pkgdir}/etc/rc.d/salt-minion
-	cp ${srcdir}/salt-syndic.initd ${pkgdir}/etc/rc.d/salt-syndic
-	chmod +x ${pkgdir}/etc/rc.d/salt-master
-	chmod +x ${pkgdir}/etc/rc.d/salt-minion
-	chmod +x ${pkgdir}/etc/rc.d/salt-syndic
-
-	mkdir -p ${pkgdir}/etc/salt/
-	cp ${srcdir}/${pkgname}-${pkgver}/conf/master ${pkgdir}/etc/salt/
-	cp ${srcdir}/${pkgname}-${pkgver}/conf/minion ${pkgdir}/etc/salt/
+    cd "$builddir"
+    python2 setup.py install --root=$pkgdir/ --optimize=1 || return 1
+    mkdir -p ${pkgdir}/var/log/$pkgname || return 1
 }
 
-md5sums="27a9bb34c205ffe07e95b316e2131177  salt-2016.3.2.tar.gz
-0b40fd1b49ff4c2f34382fd69c341c34  salt-master.initd
+init_copy() {
+    type="$1"
+    mkdir -p ${subpkgdir}/etc/init.d/ || return 1
+    mkdir -p ${subpkgdir}/etc/conf.d/ || return 1
+
+    cp ${srcdir}/salt-$type.initd ${subpkgdir}/etc/init.d/salt-$type || return 1
+    cp ${srcdir}/salt-$type.confd ${subpkgdir}/etc/conf.d/salt-$type || return 1
+
+    chmod +x ${subpkgdir}/etc/init.d/salt-$type || return 1
```

Comment:
These lines:

```sh
mkdir -p ${subpkgdir}/etc/init.d/ || return 1
cp ${srcdir}/salt-$type.initd ${subpkgdir}/etc/init.d/salt-$type
chmod +x ${subpkgdir}/etc/init.d/salt-$type
```

should be replaces by more convenient `install -m 755 -D "$srcdir"/salt-$type.initd "$subpkgdir"/init.d/salt-$type`.

The same applies for confd.